ecdsa/src/hazmat.rs

```diff
@@ -100,10 +100,7 @@ where
         // Compute 𝒔 as a signature over 𝒓 and 𝒛.
         let s = k_inv * (z + (r * self));
 
-        if s.is_zero().into() {
-            return Err(Error::new());
-        }
-
+        // NOTE: `Signature::from_scalars` checks that both `r` and `s` are non-zero.
```

Comment:
@fjarri perhaps you could add a comment like this in https://github.com/RustCrypto/elliptic-curves/pull/907 ?